R/ml_clustering_kmeans.R

```diff
@@ -148,7 +148,16 @@ new_ml_kmeans_model <- function(jobj) {
 }
 
 new_ml_kmeans_summary <- function(jobj) {
-  new_ml_clustering_summary(
+  sc <- spark_connection(jobj)
+  version <- spark_version(sc)
+
+  kmeans_summary <- new_ml_clustering_summary(
     jobj,
     class = "ml_kmeans_summary")
+
+  if (version >= "2.4.0") {
```

Comment:
Similar here, we could have `is_spark_2_4()` and remove `sc` and `version`.

Reply:
We could also consider refactoring `is_spark_3()` and `is_spark_2_4()` into `is_spark("3.0")` and `is_spark("2.4")`.